packages/collabswarm/src/collabswarm-document.ts

```diff
@@ -78,29 +79,31 @@ export class CollabswarmDocument<
     return this._document;
   }
 
-  private _hashes = new Set<string>();
+  // Document readers ACL.
+  private _readers = this._aclProvider.initialize();
+
+  // Document writers ACL.
+  private _writers = this._aclProvider.initialize();
```

Comment:
if both _readers_ and _writers_ are the return of the same function, why not just have one data member? Or I'd expect `.initialize().readers` or something

Reply:
Ah, the reason here is that the ACLProvider returns an ACL implementation which is really just a set of users. I was thinking that there are 2 ACLs that each need to be initialized this is also making the claim that the internal API of an ACL is the same regardless whether or not its a reader or a writer

Reply:
Ah, two separate instances of the same class.